src/freenet/support/Fields.java

```diff
@@ -924,21 +934,33 @@ public static int compareObjectID(Object o1, Object o2) {
 		return 0;
 	}
 	
-	/** Avoid issues with overflow, 2's complement. E.g. 0-Integer.MIN_VALUE = Integer.MIN_VALUE-0. */
+	/**
+	 * Avoid issues with overflow, 2's complement. E.g. 0-Integer.MIN_VALUE = Integer.MIN_VALUE-0.
+	 * @deprecated Use {@link Integer#compare(int, int)} */
+	@Deprecated
 	public static final int compare(int x, int y) {
 		if(x > y) return 1;
 		if(y > x) return -1;
 		return 0;
 	}
 	
-	/** Avoid issues with overflow, 2's complement. */
+	/**
+	 * Avoid issues with overflow, 2's complement.
+	 * @deprecated Use {@link Long#compare(long, long)} */
+	@Deprecated
 	public static final int compare(long x, long y) {
 		if(x > y) return 1;
 		if(y > x) return -1;
 		return 0;
 	}
 
-	/** Avoid issues with NaN's. */
+	/**
+	 * Avoid issues with NaN's.
+	 * @deprecated
+	 *     Use {@link Double#compare(double, double)}. Notice: It may behave differently with
+	 *     regards to special values (e.g. -0.0). Check its source code if you need precisely
+	 *     the same behavior. */
```

Comment:
We really can't use this kind of comments. This should either document what is different exactly. If the differences are too great, this method serves a good purpose of preventing duplication of logic and should not be deprecated.

That said, this method is blatantly inconsistent with `Double.compare` on negative integers and `NaN`.